File: src/main/java/kitchenpos/menus/tobe/menugroup/domain/model/MenuGroupV2.java
package kitchenpos.menus.tobe.menugroup.domain.model;

import java.util.UUID;
import javax.persistence.Column;
import javax.persistence.Embedded;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;
import kitchenpos.common.tobe.domain.DisplayedName;

@Table(name = "menu_group")
@Entity
public class MenuGroupV2 {

    @Column(name = "id", columnDefinition = "varbinary(16)")
    @Id
    private UUID id;

    @Embedded
    private DisplayedName name;

    protected MenuGroupV2() {
    }

    public MenuGroupV2(final DisplayedName name) {
        this.id = UUID.randomUUID();
        this.name = name;
    }

    public UUID getId() {
        return id;
    }

    public DisplayedName getName() {
        return name;
    }
}

File: src/main/kotlin/kitchenpos/tobe/menu/domain/entity/MenuGroupV2.kt
package kitchenpos.tobe.menu.domain.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.util.*

@Entity
@Table(name = "menu_group")
class MenuGroupV2(
    @Column(name = "id", columnDefinition = "binary(16)")
    @Id
    val id: UUID,
    @Column(name = "name", nullable = false)
    val name: String,
) {
    companion object {
        fun of(name: String): MenuGroupV2 {
            return MenuGroupV2(
                id = UUID.randomUUID(),
                name = name,
            )
        }
    }
}
